src/inference/gibbs.jl

```diff
@@ -20,19 +22,24 @@ end
 # Use PG for a 'v2' variable, and use HMC for the 'v1' variable.
 # Note that v2 is discrete, so the PG sampler is more appropriate
 # than is HMC.
-alg = Gibbs(HMC(0.2, 3, :v1), PG(20, 1, :v2))
+alg = Gibbs(HMC(0.2, 3, :v1), PG(20, :v2))
 ```
 
 Tips:
 - `HMC` and `NUTS` are fast samplers, and can throw off particle-based
 methods like Particle Gibbs. You can increase the effectiveness of particle sampling by including
 more particles in the particle sampler.
 """
-mutable struct Gibbs{A} <: InferenceAlgorithm
-    algs      ::  A   # component sampling algorithms
-    function Gibbs(algs...)
-        return new{typeof(algs)}(algs)
-    end
+struct Gibbs{space, A<:Tuple{Vararg{GibbsComponent}}} <: InferenceAlgorithm
+    algs::A   # component sampling algorithms
+end
+
+function Gibbs(algs::GibbsComponent...)
+    # obtain space of sampling algorithms
+    spacearray = collect(Iterators.flatten(getspace(alg) for alg in algs))::Vector{Symbol}
```

Comment:
`Tuple(union(getspace.(algs)...))` is shorter and more readable.

Reply:
That one is nice! It yields a slightly different result with duplicates being removed, but I guess that's fine here (since we do not throw an error in case of duplicates anyway). I tried to not use the splatting operator, but the more readable version seems to be even slightly faster for some simple examples I tried :+1: